From ca3d32ef278d99632103883d570e73cd96af747b Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Mon, 26 Dec 2016 16:28:48 +0200
Subject: [PATCH 0658/1345] dts: a8040: a7040: add comphy handles to sata
 nodes

commit  3ec3550576923784e41d3bc0221b5c5537b1398e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch adds comphy handles to SATA nodes, this allows to
configure comphy into SATA mode in Linux Kernel
(using the comphy driver - no need to rely on u-boot configuration).

Change-Id: I0fed3d2c60acb5abddd396b243268db8fbdd3667
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34813
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../arm64/boot/dts/marvell/armada-7040-db-nand.dts |    2 ++
 .../boot/dts/marvell/armada-7040-db-router.dts     |    2 ++
 .../boot/dts/marvell/armada-70x0-customer.dts      |    4 ++++
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts   |   10 +++++++++-
 .../boot/dts/marvell/armada-8040-db-router.dts     |    4 ++++
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts  |    6 ++++++
 .../boot/dts/marvell/armada-80x0-customer.dts      |    8 ++++++++
 7 files changed, 35 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
index 534eaa0..ea95f23 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
@@ -92,6 +92,8 @@
 					status = "disabled";
 				};
 				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "okay";
 				};
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
index 451bb21..eeed1f6 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
@@ -92,6 +92,8 @@
 					status = "disabled";
 				};
 				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "okay";
 				};
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
index dd148bf..41288fd 100644
--- a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
@@ -131,10 +131,14 @@
 
 				sata-port@0 {
 					/* UPDATE */
+					phys = <&comphy0 1 COMPHY_SATA0>;
+					phy-names = "comphy";
 					status = "disabled";
 				};
 				sata-port@1 {
 					/* UPDATE */
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "disabled";
 				};
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index 36f1e9d..f1ac384 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -79,9 +79,13 @@
 				status = "okay";
 
 				sata-port@0 {
-					satus = "okay";
+					phys = <&comphy0 1 COMPHY_SATA0>;
+					phy-names = "comphy";
+					status = "okay";
 				};
 				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "okay";
 				};
 			};
@@ -158,9 +162,13 @@
 				status = "okay";
 
 				sata-port@0 {
+					phys = <&comphy1 1 COMPHY_SATA0>;
+					phy-names = "comphy";
 					status = "okay";
 				};
 				sata-port@1 {
+					phys = <&comphy1 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "okay";
 				};
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
index dd4aace..0875114 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
@@ -114,6 +114,8 @@
 					satus = "disabled";
 				};
 				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "okay";
 				};
 			};
@@ -189,6 +191,8 @@
 					satus = "disabled";
 				};
 				sata-port@1 {
+					phys = <&comphy1 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "okay";
 				};
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index c59d347..656f3e6 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -198,6 +198,8 @@
 		status = "disabled";
 	};
 	sata-port@1 {
+		phys = <&comphy0 5 COMPHY_SATA1>;
+		phy-names = "comphy";
 		status = "okay";
 	};
 };
@@ -219,8 +221,12 @@
 
 	sata-port@0 {
 		status = "okay";
+		phys = <&comphy1 1 COMPHY_SATA0>;
+		phy-names = "comphy";
 	};
 	sata-port@1 {
+		phys = <&comphy1 3 COMPHY_SATA1>;
+		phy-names = "comphy";
 		status = "okay";
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
index e178aa1..5f2301c 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
@@ -140,10 +140,14 @@
 
 				sata-port@0 {
 					/* UPDATE */
+					phys = <&comphy0 1 COMPHY_SATA0>;
+					phy-names = "comphy";
 					status = "disabled";
 				};
 				sata-port@1 {
 					/* UPDATE */
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "disabled";
 				};
 			};
@@ -255,10 +259,14 @@
 
 				sata-port@0 {
 					/* UPDATE */
+					phys = <&comphy1 1 COMPHY_SATA0>;
+					phy-names = "comphy";
 					status = "disabled";
 				};
 				sata-port@1 {
 					/* UPDATE */
+					phys = <&comphy1 3 COMPHY_SATA1>;
+					phy-names = "comphy";
 					status = "disabled";
 				};
 			};
-- 
1.7.9.5

